block-storage/src/main/scala/coop/rchain/blockstorage/dag/BlockDagStorage.scala

```diff
@@ -37,6 +39,8 @@ trait BlockDagRepresentation[F[_]] {
       startBlockNumber: Long,
       maybeEndBlockNumber: Option[Long]
   ): F[Vector[Vector[BlockHash]]]
+  // DAG representation has to have finalized block, or it does not make sense
+  def lastFinalisedBlock: BlockHash
```

Comment:
In the end, I decided to have LFB be present without Option. It required more work but I believe it worth it.